src/figuresClasses.ts

```diff
@@ -1,19 +1,100 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+function isTriangle(side1: number, side2: number, side3: number): boolean {
+  const sides: number[] = [side1, side2, side3];
+  const maxSide: number = Math.max(side1, side2, side3);
+
+  const twoSides = sides.filter((side: number) => side !== maxSide);
+
+  const sumTwoSides = twoSides[0] + twoSides[1];
+
+  if (maxSide < sumTwoSides) {
+    return false;
+  }
 
+  return true;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('the number is not positive');
+    }
+
+    if (isTriangle(this.a, this.b, this.c)) {
+      throw new Error('is not a triangle');
+    }
+
+    this.color = color;
```

Comment:
The properties color, a, b, c are already initialized due to public keyword in constructor, no need to reassign them.